BUILDING_CALICO.md

```diff
@@ -74,6 +74,24 @@ export VERSION_CALICO=v2.3.0
 
 ```
 
+#### Build targets
+The following are the standard `Makefile` targets that are in every project repository.
+
+* `make build`: build the binary for the current architecture. Normally will be in `bin/` or `dist/` and named `NAME-ARCH`, e.g. `felix-arm64` or `typha-amd64`. If there are multiple OSes available, then named `NAME-OS-ARCH`, e.g. `calicoctl-darwin-amd64`.
+* `make build ARCH=<ARCH>`: build the binary for the given `ARCH`. Output binary will be in `bin/` or `dist/` and follows the naming convention listed above.
+* `make build-all`: build binaries for all supported architectures. Output binaries will be in `bin/` or `dist/` and follow the naming convention listed above.
+* `make image`: create a docker image for the current architecture. It will be named `NAME:latest-ARCH`, e.g. `calico/felix:latest-amd64` or `calico/typha:latest-s390x`. If multiple operating systems are available, will be named `NAME:latest-OS-ARCH`, e.g. `calico/ctl:latest-linux-ppc64le`
+* `make image ARCH=<ARCH>`: create a docker image for the given `ARCH`. Images will be named according to the convention listed above.
+* `make image-all`: create docker images for all supported architectures. Images will be named according to the convention listed above in `make image`.
+* `make push IMAGETAG=<IMAGETAG>`: push the docker image for the current architecture to all registries, specifically docker hub and quay.io. The images will be named `NAME:IMAGETAG-ARCH`, e.g. `make push IMAGENAME=foo` will push images `calico/felix:foo-amd64` or `calico/typha:foo-ppc64le`. 
```

Comment:
I think we probably want to leave the `push` targets out for now since those are primarily for releases and aren't standardized yet AFAIK.

The rest looks really nice and will be great for people looking to build Calico :)

Reply:
IMO this one is very important and helps to push images easily in semaphore like jobs once tests are done!

Reply:
Is this intended to be the desired set of targets, or do document the current state? Most of the projectcalico repos don't have a `make push` target AFAICT. 

Agreed it's a nice to have shortcut. The CNI plugin and kube-controllers repos both have `make release-publish` which is intended to do that for a release build. Would this be something different?

Reply:
> I think we probably want to leave the push targets out for now since those are primarily for releases and aren't standardized yet AFAIK
> Is this intended to be the desired set of targets, or to document the current state? Most of the projectcalico repos don't have a make push target AFAICT

Desired set. I just updated the first PR, based on the conversations with @fasaxc , to include `make push`. See [this line](https://github.com/projectcalico/felix/pull/1746/files#diff-b67911656ef5d18c4ae36cb6741b7965R304) in [this felix PR](https://github.com/projectcalico/felix/pull/1746)

> The CNI plugin and kube-controllers repos both have make release-publish which is intended to do that for a release build. Would this be something different?

So do some others have that target. It is something different, although it took some explaining by @fasaxc for me to understand it.

* `make push IMAGETAG=<tag>` releases a given architecture (current or override if specified `ARCH=<arch>`) for a certain `<tag>`. It might be an actual release, might be `master` (used in `calico/ctl`, I believe), it might be a hash, it might be anything.
* `make release-publish` publishes a specific released version, including rpms, docker images, etc. It will call `make push IMAGETAG=$(VERSION)`